modules/tensor_mechanics/src/materials/ComputeDamageStress.C

```diff
@@ -0,0 +1,52 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeDamageStress.h"
+#include "DamageBase.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeDamageStress);
+
+template <>
+InputParameters
+validParams<ComputeDamageStress>()
+{
+  InputParameters params = validParams<ComputeFiniteStrainElasticStress>();
+  params.addClassDescription(
+      "Compute stress for damaged elastic materials in conjunction with a damage model.");
+  params.addRequiredParam<MaterialName>("damage_model", "Name of the damage model");
+  return params;
+}
+
+ComputeDamageStress::ComputeDamageStress(const InputParameters & parameters)
+  : ComputeFiniteStrainElasticStress(parameters), _damage_model(nullptr)
+{
+}
+
+void
+ComputeDamageStress::initialSetup()
+{
+  MaterialName damage_model_name = getParam<MaterialName>("damage_model");
+  DamageBase * dmb = dynamic_cast<DamageBase *>(&getMaterialByName(damage_model_name));
+  if (dmb)
+    _damage_model = dmb;
+  else
+    mooseError("Damage Model " + damage_model_name + " is not compatible with ComputeDamageStress");
```

Comment:
I suggest using `paramError` here:

```
paramError("damage_model", "Damage Model " + damage_model_name + " is not compatible with ComputeDamageStress"); 
```

Reply:
I think mooseError is more appropriate here because paramError gives this message that doesn't really make sense:
```
*** ERROR ***
param 'Damage Model damage_index is not compatible with ComputeDamageStress' not present in InputParams
```

Reply:
mooseError gives this:
```
*** ERROR ***
Damage Model damage_index is not compatible with ComputeDamageStress
```
which is what I wanted.

Reply:
:-D that's because you forgot to add the additional first argument!